metecho/api/serializers.py

```diff
@@ -158,6 +158,11 @@ class Meta:
             GitHubUserValidator(parent="repository"),
         )
 
+    def create(self, validated_data):
+        instance = super().create(validated_data)
+        instance.create_gh_branch(self.context["request"].user)
+        return instance
```

Comment:
@wlonk This seems to bypass the `CaseInsensitiveUniqueTogetherValidator` when an existing project name is used, now triggering a `django.db.utils.IntegrityError: duplicate key value violates unique constraint "api_project_name_product_id_2b3ad4de_uniq"` instead of the prior `400` response with user-friendly message.

Reply:
@wlonk Note from our mtg: Let's revisit this (and all) unique/unique-together validators, to be sure they're handling soft-deleted models appropriately.

Reply:
Turns out I had written the `CaseInsensitiveUniqueTogetherValidator` to handle soft deletes, but I was still enforcing a DB-level unique-together, thus causing the error you were seeing with a DB integrity error rising to the frontend.

I'll remove that constraint, just takes a moment hand-testing.